src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/source_filters_table/components/table/table.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiInMemoryTable,
+  EuiFieldText,
+  EuiButtonIcon,
+  keyCodes,
+  RIGHT_ALIGNMENT,
+} from '@elastic/eui';
+
+export class Table extends Component {
+  static propTypes = {
+    indexPattern: PropTypes.object.isRequired,
+    items: PropTypes.array.isRequired,
+    deleteFilter: PropTypes.func.isRequired,
+    fieldWildcardMatcher: PropTypes.func.isRequired,
+    saveFilter: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      editingFilterId: null,
+      editingFilterValue: null,
+    };
+  }
+
+  startEditingFilter = (id, value) => this.setState({ editingFilterId: id, editingFilterValue: value })
+  stopEditingFilter = () => this.setState({ editingFilterId: null })
+  onEditingFilterChange = e => this.setState({ editingFilterValue: e.target.value })
+
+  onEditFieldKeyPressed = ({ charCode }) => {
+    if (keyCodes.ENTER === charCode) {
+      this.props.saveFilter({
+        filterId: this.state.editingFilterId,
+        newFilterValue: this.state.editingFilterValue,
+      });
+      this.stopEditingFilter();
+    }
+    if (keyCodes.ESCAPE === charCode) {
+      this.stopEditingFilter();
+    }
+  }
+
+  getColumns() {
+    const {
+      deleteFilter,
+      fieldWildcardMatcher,
+      indexPattern,
+      saveFilter,
+    } = this.props;
+
+    return [
+      {
+        field: 'value',
+        name: 'Filter',
+        description: `Filter name`,
+        dataType: 'string',
+        sortable: true,
+        render: (value, filter) => {
+          if (this.state.editingFilterId === filter.clientId) {
+            return (
+              <EuiFieldText
+                value={this.state.editingFilterValue}
+                onChange={this.onEditingFilterChange}
+                onKeyPress={this.onEditFieldKeyPressed}
```

Comment:
Can we change `onKeyPress` to `onKeyDown`? This also makes the handler execute more quickly because we don't wait for the user to release the key.

Reply:
👍